### PR TITLE
Section on using LoRA alpha / scale

### PR DESCRIPTION
**Pull request by pcuenca**

### CODE REVIEW

File: docs/source/en/training/lora.mdx [hunk 1]
```diff
@@ -150,6 +150,27 @@ This is especially useful when you don't want to hardcode the base model identif
...
+pipe = StableDiffusionPipeline.from_pretrained("CompVis/stable-diffusion-v1-4", torch_dtype=torch.float16)
+pipe.unet.load_attn_procs(model_path)
+pipe.to("cuda")
+
+prompt = "A pokemon with blue eyes."
+image = pipe(prompt, num_inference_steps=30, guidance_scale=7.5, cross_attention_kwargs={"scale": 0.5}).images[0]
+image.save("pokemon.png")
+```
+
+A value of `0` is the same as _not_ using the LoRA weights, whereas `1` means only the fine-tuned weights will be used. Values between 0 and 1 will merge the two versions accordingly.
```
Comment (sayakpaul) [inline docs/source/en/training/lora.mdx:171]: I guess it removes the ambiguity. 

```suggestion
A value of `0` is the same as _not_ using the LoRA weights, whereas `1` means only the LoRA fine-tuned weights will be used. Values between 0 and 1 will merge the two versions accordingly.
```

File: docs/source/en/training/lora.mdx [hunk 2]
```diff
@@ -150,6 +150,27 @@ This is especially useful when you don't want to hardcode the base model identif
 Inference for DreamBooth training remains the same. Check
 [this section](https://github.com/huggingface/diffusers/tree/main/examples/dreambooth#inference-1) for more details. 
 
+### Merging LoRA with original model
+
+You can adjust the merging ratio between the fine-tuned model and the original. This is controlled by a parameter called α (alpha) in the paper, or `scale` in our implementation. You can tweak it with the following code, that passes `scale` as `cross_attention_kwargs` in the pipeline call:
+
```
Comment (sayakpaul) [inline docs/source/en/training/lora.mdx:156]: Maybe add a preamble stating that during inference, LoRA weights are merged with the frozen pre-trained weights and then give a small example?

**Review by sayakpaul — CHANGES_REQUESTED** [review]
Thanks!

Just minor two things.

**Review by patil-suraj — APPROVED** [review]
Looks good, thanks!

**Review by patrickvonplaten — APPROVED** [review]
Thanks!

File: docs/source/en/training/lora.mdx (hunk 2 above)
Comment (pcuenca) [reply docs/source/en/training/lora.mdx:156]: Well, this is inside the inference section already :) I thought that all references to "merging" were clear, but I'll reword a little bit. I'm not sure what you mean by a "small example", do you mean an image strip?

Comment (sayakpaul) [reply docs/source/en/training/lora.mdx]: Yeah, maybe an image strip showing two weight matrices being merged (I think the LoRA paper already has something similar). Or maybe an equation to make it explicitly clear. No strong opinions, though.

**Comment by KadriMufti** [comment]
Is there any advice on how to choose an appropriate scaling factor (alpha)? 

In the paper the authors said they chose it to be the same as the rank they used, and they do not tune it. I saw in some demo/tutorial notebooks that people are choosing alpha to be 2x and 4x the chosen rank size.

**Comment by sayakpaul** [comment]
Pinging our in-house expert @pacman100 for advice.

**Comment by pacman100** [comment]
Hello @KadriMufti, as you mentioned, it is typically 4X or 2X of `r`.  To provide more data points, [AdaLoRA](https://arxiv.org/pdf/2303.10512.pdf) paper mentions. `Typically α is set as 16 or 32 and never tuned`.  In [MAM](https://arxiv.org/pdf/2110.04366.pdf) paper, they found LoRA with alpha being 4X to that of rank to perform the best. They also use a scaling factor of 4X for their Parallel Adapter variant too which is inspired by the compositional form of LoRA. **It just means that weighting the delta activations higher helps in tuning the lora weights**. So, 4X or 2X of `r` is a good starting point as has been empirically seen in the papers.

Screenshot 2023-04-06 at 10 32 45 AM

**Pull request merged**